src/components/Common/Login/ResetPasswordForm.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <form id="resetPasswordForm" method="post" @submit.prevent="resetPassword()">
+    <div class="resetPasswordForm-inputs">
+      <b-form-group
+        label="New password"
+        label-for="password"
+        label-cols-sm="4"
+        label-cols-md="3"
+      >
+        <b-form-input
+          autofocus
+          class="validate"
+          data-cy="ResetPassword-password"
+          id="password"
+          name="password"
+          required
+          tabindex="1"
+          type="password"
+          v-model="password"
+        />
+      </b-form-group>
+
+      <b-form-group
+        description="Re-type the password for confirmation"
+        label="Confirm password"
+        label-cols-sm="4"
+        label-cols-lg="3"
+        label-for="password2"
+      >
+        <b-input
+          class="validate"
+          data-cy="ResetPassword-password2"
+          id="password2"
+          name="password2"
+          required
+          tabindex="2"
+          type="password"
+          v-model="password2"
+        />
+      </b-form-group>
+
+      <div class="ResetPasswordForm-error" v-if="error">
+        <b-alert variant="danger" show dismissible>
+          Error: {{ error }}
+        </b-alert>
+      </div>
+
+      <div class="ResetPasswordForm-buttons float-right mt-3">
+        <b-button
+          variant="primary"
+          data-cy="ResetPassword-submitBtn"
+          type="submit"
+          name="action"
+          tabindex="3"
+          >Send</b-button
+        >
+      </div>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'ResetPasswordForm',
+  props: {
+    resetToken: String
+  },
+  data() {
+    return {
+      error: '',
+      password: '',
+      password2: ''
+    }
+  },
+  methods: {
+    async resetPassword() {
+      this.error = ''
+
+      if (this.password.trim() === '' || this.password2.trim() === '') {
+        this.error = 'All fields are mandatory'
+        return
+      }
+
+      if (this.password !== this.password2) {
+        this.error = 'Passwords do not match'
+        return
+      }
+
+      try {
+        await this.$store.direct.dispatch.auth.doResetPassword({
+          password: this.password,
+          token: this.$route.params.token
+        })
+        this.$emit('reset-password::after')
+      } catch (error) {
+        this.error = error.message
+      }
```

Comment:
We probably should not display the kuzzle error returned since it might be not enough human readable.
To be consistent with the others catch blocks of the AC we should log that error in the console and display a toast
```suggestion
        this.error = 'An error occured.'
        console.error(error.message);
        this.$bvToast.toast(
            'The complete error has been printed to the console.',
            {
              title: err.message,
              variant: 'danger',
              toaster: 'b-toaster-bottom-right',
              appendToast: true
            }
          )
      }
```

Reply:
There are some cases where the Kuzzle error is just fine. Ben showed some screenshots where the errors where readable enough

Reply:
I only cought this behaviour on the signup component, but I believe it should be changed everywhere.

Error messages (not only Kuzzle ones) are supposed to be human-readable. 

I can only see situations where the users should get it:

- Either the message is useful to him, like a validation error, to fix the situation
- Or the error cannot be fixed by the user (bug, configuration error, third party down etc.), in which case the error message should be reported to the support.

Asking the user to open the development console to get it is probably a blocker for some users.

Reply:
Yes, the best practice should be:
* intercept the error and parse the error code
* display a toast with a human-readable message
* report the original Kuzzle error to the console, so that the user can report it to the support team if necessary